lib/screens/profile_screen.dart

```diff
@@ -0,0 +1,174 @@
+import 'error_screen.dart';
+
+class ProfileScreen extends StatefulWidget {
+  @override
+  State createState() {
+    return ProfileScreenState();
+  }
+}
+
+class ProfileScreenState extends State<ProfileScreen> {
+  String preferredSourceLanguage = "";
+
+  @override
+  void initState() {
+    _getPreferredSourceLanguage().then((value) {
+      setState(() {
+        preferredSourceLanguage = value ?? defaultSourceLang;
+      });
+    });
+    super.initState();
+  }
+
+  Future<User> _getProfileInfoFromPreferences() async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+
+    return User(
+        username: prefs.getString("username"),
+        firstName: prefs.getString("firstName"),
+        lastName: prefs.getString("lastName"),
+        location: prefs.getString("location"),
+        avatar: prefs.getString("avatar"));
+  }
+
+  Future<String> _getPreferredSourceLanguage() async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+    return prefs.getString("preferredSourceLanguage");
+  }
+
+  void _setPreferredSourceLang(String preferredLang) async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+    prefs.setString("preferredSourceLanguage", preferredLang);
+
+    setState(() {
+      preferredSourceLanguage = preferredLang;
+    });
+  }
+
+  void _doLogout() async {
+    final FacebookLogin facebookSignIn = new FacebookLogin();
+    if (await facebookSignIn.isLoggedIn) {
+      await facebookSignIn.logOut();
+
+      while (Navigator.canPop(context)) {
+        Navigator.pop(context);
+      }
+    }
+  }
+
+  void _changeSourceLanguage() {
+    Widget languages = FutureBuilder(
+      future: fetchLanguages("type=foreign"),
+      builder: (context, snapshot) {
+        if (snapshot.connectionState == ConnectionState.none) {
+          return ErrorScreen(errorType.noConnection);
+        }
+        if (snapshot.hasError) {
+          return ErrorScreen(errorType.exception);
+        }
+        if (snapshot.hasData) {
+          return ListView.builder(
+              itemCount: snapshot.data.length,
+              itemBuilder: (context, index) {
+                return InkWell(
+                    onTap: () {
+                      _setPreferredSourceLang(snapshot.data[index].name);
+                      Navigator.pop(context);
+                    },
+                    highlightColor: Theme.of(context).primaryColorLight,
+                    child: ListTile(
+                      title: Text(snapshot.data[index].name),
+                      subtitle: Text(snapshot.data[index].code),
+                    ));
+              });
+        } else {
+          return Center(child: CircularProgressIndicator());
+        }
+      },
+    );
+
+    showDialog<void>(
+        context: context,
+        barrierDismissible: true,
+        builder: (BuildContext context) {
+          return AlertDialog(
+              title: Text("Change Language"), content: languages);
```

Comment:
not a big deal but probably want to localize this string.

Reply:
Will do that